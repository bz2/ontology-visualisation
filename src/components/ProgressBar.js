import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next'

const ProgressBar = ({
  progress
}) => {
  const { t } = useTranslation()
  const width = `${progress * 2}px`

  return (
    <div className="progress-bar">
      <div className="progress-bar-title">
        {t('loading')}
      </div>

      <div className="progress-bar-box">
        <div
          className="progress-bar-box-filler"
          style={{ width }}
        />
      </div>
    </div>
  )
}

ProgressBar.propTypes = {
  progress: PropTypes.number.isRequired,
}

export default ProgressBar
