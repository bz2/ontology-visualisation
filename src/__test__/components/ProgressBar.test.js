import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ProgressBar from '../../components/ProgressBar'

const setup = () => {
  const props = {
    progress: 10
  }

  const component = shallow(<ProgressBar {...props} />)

  return {
    component,
    props
  }
}

describe('ProgressBar', () => {
  it('should match snapshot ', () => {
    const {
      component
    } = setup()

    expect(toJson(component)).toMatchSnapshot()
  })
})
