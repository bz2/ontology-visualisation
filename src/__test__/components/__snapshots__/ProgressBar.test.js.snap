// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should match snapshot  1`] = `
<div
  className="progress-bar"
>
  <div
    className="progress-bar-title"
  >
    loading
  </div>
  <div
    className="progress-bar-box"
  >
    <div
      className="progress-bar-box-filler"
      style={
        Object {
          "width": "20px",
        }
      }
    />
  </div>
</div>
`;
